Signal/src/view controllers/MessagesViewController.h

```diff
@@ -14,6 +14,24 @@
 
 extern NSString *const OWSMessagesViewControllerDidAppearNotification;
 
+@interface OWSMessagesComposerTextView : JSQMessagesComposerTextView
+
+@end
+
+#pragma mark -
+
+@interface OWSMessagesToolbarContentView : JSQMessagesToolbarContentView
+
+@end
+
+#pragma mark -
+
+@interface OWSMessagesInputToolbar : JSQMessagesInputToolbar
+
+@end
+
+#pragma mark -
+
 @interface MessagesViewController : JSQMessagesViewController <UIImagePickerControllerDelegate,
```

Comment:
To support paste, I had to use a custom text view.  `JSQMessagesViewController` doesn't make it easy to use a custom text view; I had to subclass a few JSQ classes AND clone a few JSQ .xibs, etc.